### PR TITLE
New Rest Route To Get Corresponding Vendors Product Categories Under StoreController

### PR DESCRIPTION
**Pull request by nurul-umbhiya**
new: added a new method named get_published_products() under Vendor class to get all published products of a vendor
new: added a new method named get_store_categories() under Vendor class to get all published products assigned categories
new: added get_best_selling_products() method under Vendor class
new added best selling products category support under get_store_categories() in Vendor class

update: rewritten store category dokan_store_category_menu() function

REST API:
new: added a new rest route name id/categories under StoreController to get corresponding vendors published product categories (wp-json/dokan/v1/stores/3/categories)
new: added popular category support under StoreController rest API (wp-json/dokan/v1/stores/3/categories?best_selling=1)

### CODE REVIEW

**Comment by nurul-umbhiya** [comment]
@shanjidah kindly consult with me before testing this PR

**Comment by FaraziF** [comment]
On the vendor store page, display the deleted categories from the store product category.

**Comment by nurul-umbhiya** [comment]
@FaraziF bhai fixed.

File: includes/REST/StoreController.php [hunk 1]
```diff
@@ -895,4 +917,29 @@ public function batch_update( $request ) {
 
         return $response;
     }
+
+    /**
+     * Get singe store
+     *
+     * @param $request
+     *
+     * @since 3.2.11
```
Comment (xaviranik) [inline includes/REST/StoreController.php:926]: Kindly replace with `DOKAN_PRO_SINCE`

File: includes/Vendor/Vendor.php [hunk 2]
```diff
@@ -528,6 +528,139 @@ public function get_products() {
         return $products;
     }
 
+    /**
+     * Get a vendor all published products
+     *
+     * @since 3.2.11
```
Comment (xaviranik) [inline includes/Vendor/Vendor.php:534]: Kindly replace with DOKAN_PRO_SINCE

File: includes/Vendor/Vendor.php [hunk 3]
```diff
@@ -528,6 +528,139 @@ public function get_products() {
...
+            set_transient( $transient_key, $products, WEEK_IN_SECONDS );
+        }
+
+        return $products;
+    }
+
+    /**
+     * Get a vendor all published products
+     *
+     * @since 3.2.11
```
Comment (xaviranik) [inline includes/Vendor/Vendor.php:557]: Kindly replace for all `@since` with `DOKAN_PRO_SINCE`

**Review by xaviranik — CHANGES_REQUESTED** [review]

**Review by xaviranik — APPROVED** [review]

**Pull request merged**